src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs

```diff
@@ -193,9 +197,19 @@ .. call.BlockStateCalls.Select(b => (long)(b.BlockOverrides?.Number ?? ulong.Min
         SimulateOutput results = _blockchainBridge.Simulate(header, tx, token);
 
         if (results.Error is not null && (results.Error.Contains("invalid transaction")
-                                      || results.Error.Contains("InsufficientBalanceException")))
+                                      || results.Error.Contains("InsufficientBalanceException")
+                                      ))
             results.ErrorCode = ErrorCodes.InvalidTransaction;
 
+        if (results.Error is not null && results.Error.Contains("InvalidBlockException"))
+            results.ErrorCode = ErrorCodes.InvalidParams;
+
+
+        if (results.Error is not null && results.Error.Contains("below intrinsic gas"))
+            results.ErrorCode = ErrorCodes.InsufficientIntrinsicGas;
```

Comment:
Can we pass the ErrorCode explicitly rather than parsing string?

Reply:
Sadly "InvalidBlockException" happens for multiple cases here and differs only by its text currently...